libcodechecker/libhandlers/cmd.py

```diff
@@ -309,7 +330,227 @@ def __register_suppress(parser):
+                                  "the server's <CONFIG_DIRECTORY>.")
+
+        dbmodes.add_argument('--postgresql',
+                             dest="postgresql",
+                             action='store_true',
+                             required=False,
+                             default=argparse.SUPPRESS,
+                             help="Specifies that a PostgreSQL database is "
+                                  "to be used instead of SQLite. See the "
+                                  "\"PostgreSQL arguments\" section on how "
+                                  "to configure the database connection.")
+
+        PGSQL_PRODUCT_NAME_DEFAULT_VAR = '<PRODUCT_NAME>'
+        pgsql = parser.add_argument_group(
+            "PostgreSQL arguments",
+            "Values of these arguments are ignored, unless '--postgresql' is "
+            "specified! The database specified here must exist, and be "
+            "connectible by the server.")
+
+        # TODO: --dbSOMETHING arguments are kept to not break interface from
+        # old command. Database using commands such as "CodeChecker store" no
+        # longer supports these --- it would be ideal to break and remove args
+        # with this style and only keep --db-SOMETHING.
+        pgsql.add_argument('--dbaddress', '--db-host',
+                           type=str,
+                           dest="dbaddress",
+                           default="localhost",
+                           required=False,
+                           help="Database server address.")
+
+        pgsql.add_argument('--dbport', '--db-port',
+                           type=int,
+                           dest="dbport",
+                           default=5432,
+                           required=False,
+                           help="Database server port.")
+
+        pgsql.add_argument('--dbusername', '--db-username',
+                           type=str,
+                           dest="dbusername",
+                           default=PGSQL_PRODUCT_NAME_DEFAULT_VAR,
+                           required=False,
+                           help="Username to use for connection.")
+
+        pgsql.add_argument('--dbpassword', '--db-password',
+                           type=str,
+                           dest="dbpassword",
+                           default="",
+                           required=False,
+                           help="Password to use for authenticating the "
+                                "connection.")
+
+        pgsql.add_argument('--dbname', '--db-name',
+                           type=str,
+                           dest="dbname",
+                           default=PGSQL_PRODUCT_NAME_DEFAULT_VAR,
+                           required=False,
+                           help="Name of the database to use.")
+
+        def __handle(args):
+            """Custom handler for 'add' so custom error messages can be
```

Comment:
This same code resides in `server.py` too. This is because `main()` everywhere deals with a **proper** `argparse.Namespace`. This `__handle` is basically an extension over argparse's argument parsing routine, as there are some special rules that can NOT be expressed with argparse's existing tooling.